detox/src/configuration/index.js

```diff
@@ -102,8 +109,34 @@ async function composeDetoxConfig({
     globalConfig,
     localConfig,
     cliConfig,
+    isCloudSession
+  });
+
+  const cloudAuthenticationConfig = await validateCloudAuthConfig({
+    errorComposer,
+    localConfig,
+    isCloudSession
   });
 
+  if (isCloudSession) {
+    const query_param = {
+      'device': _.get(deviceConfig, 'device.name'),
+      'os': _.get(deviceConfig, 'device.os'),
+      'osVersion': _.get(deviceConfig, 'device.osVersion'),
+      'name': _.get(sessionConfig, 'name'),
+      'project': _.get(sessionConfig, 'project'),
+      'build': _.get(sessionConfig, 'build'),
+      'clientDetoxVersion': package_json.version,
+      'app': _.get(appsConfig, 'default.app'),
+      'appClient': _.get(appsConfig, 'default.appClient'),
+      'username': _.get(cloudAuthenticationConfig, 'username'),
+      'accessKey': _.get(cloudAuthenticationConfig, 'accessKey'),
+      'networkLogs': _.get(artifactsConfig, 'plugins.networkLogs.enabled'),
+      'deviceLogs': _.get(artifactsConfig, 'plugins.deviceLogs.enabled'),
+      'video': _.get(artifactsConfig, 'plugins.video.enabled')
+    };
+    sessionConfig.server += `?caps=${encodeURIComponent(JSON.stringify(query_param))}`;
```

Comment:
**[MUST HAVE]**

These params are needed on a real device elastic cloud so that all session set up can be done with minimal network calls and in a synchronous single step.